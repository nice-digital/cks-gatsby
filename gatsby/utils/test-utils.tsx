import React, { ReactElement } from "react";
import { render, RenderResult } from "@testing-library/react";
import {
	createMemorySource,
	createHistory,
	LocationProvider,
	History,
} from "@reach/router";

// Util for rendering a component wrapped within Reach Router's LocationProvider
// See https://testing-library.com/docs/example-reach-router
export const renderWithRouter = (
	ui: ReactElement,
	{ route = "/", history = createHistory(createMemorySource(route)) } = {}
): { history: History } & RenderResult => ({
	...render(<LocationProvider history={history}>{ui}</LocationProvider>),
	// adding `history` to the returned utilities to allow us
	// to reference it in our tests (just try to avoid using
	// this to test implementation details).
	history,
});
