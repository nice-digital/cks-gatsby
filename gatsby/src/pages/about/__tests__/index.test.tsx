/* eslint-disable testing-library/no-node-access */
import React from "react";
import { textContentMatcher } from "test-utils";
import { render, screen } from "@testing-library/react";

import AboutPage from "../";

describe("About page", () => {
	beforeEach(() => {
		// eslint-disable-next-line testing-library/no-render-in-setup
		render(<AboutPage />);
	});

	describe("Breadcrumbs", () => {
		it("should render NICE homepage breadcrumb link", () => {
			expect(
				screen.queryByText(textContentMatcher("NICE"), {
					selector: ".breadcrumbs a",
				})
			).toHaveAttribute("href", "https://www.nice.org.uk/");
		});

		it("should render CKS home breadcrumb", () => {
			expect(
				screen.queryByText(textContentMatcher("CKS"), {
					selector: ".breadcrumbs a",
				})
			).toHaveAttribute("href", "/");
		});

		it("should render current page breadcrumb without link", () => {
			expect(
				screen.getByText(textContentMatcher("About CKS"), {
					selector: ".breadcrumbs span",
				})
			).toBeTruthy();
		});
	});

	describe("Page header", () => {
		it("should render heading 1 with About CKS", () => {
			expect(
				screen.queryByText("About CKS", {
					selector: "h1",
				})?.tagName
			).toBe("H1");
		});
	});

	describe("Body", () => {
		it("should match snapshot for 'Help develop CKS' panel", () => {
			expect(
				screen.getByText("Help develop CKS").parentElement
			).toMatchSnapshot();
		});

		it("should match snapshot for main body content", () => {
			expect(
				screen.getByText("Quick answers to clinical questions").parentElement
			).toMatchSnapshot();
		});
	});
});
