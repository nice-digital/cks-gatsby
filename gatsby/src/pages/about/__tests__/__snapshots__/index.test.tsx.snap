// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page Body should match snapshot for 'Help develop CKS' panel 1`] = `
<div
  class="panel"
>
  <h2
    class="h3"
  >
    Help develop CKS
  </h2>
  <p>
    CKS topics rely on feedback from healthcare professionals, patients groups, patients and carers to ensure they are accurate and relevant.
  </p>
  <ul
    aria-label="How to help develop CKS"
  >
    <li>
      <a
        href="https://cks.clarity.co.uk/get-involved/"
      >
        Help develop CKS topics
      </a>
    </li>
    <li>
      <a
        href="https://www.nice.org.uk/cks-feedback"
      >
        Send feedback on CKS topics
      </a>
    </li>
  </ul>
</div>
`;

exports[`About page Body should match snapshot for main body content 1`] = `
<div
  class="body"
  data-g="12 md:7 lg:8"
>
  <h2>
    Quick answers to clinical questions
  </h2>
  <p>
    Clinical Knowledge Summaries are concise, accessible summaries of current evidence for primary care professionals. There are over 350 topics to chose from.
  </p>
  <p>
    The topics focus on the most common and significant presentations in primary care. They give trusted information to support safe decision-making and improve standards of patient care.
     
  </p>
  <h2>
    Developed on behalf of NICE
  </h2>
  <p>
    CKS topics are developed by Clarity Informatics Ltd but commissioned and funded by NICE.
  </p>
  <p>
    Topics are written by an expert multidisciplinary team with experience of primary care, supported by a network of over 6000 specialist external reviewers.
  </p>
  <p>
    The development process is
     
    <a
      href="https://www.nice.org.uk/about/what-we-do/accreditation"
    >
      accredited by NICE
    </a>
     
    to ensure the highest quality.
  </p>
  <p>
    <a
      href="/about/development/"
    >
      Read more about the CKS development process
    </a>
  </p>
  <h2>
    Evidence-based
  </h2>
  <p>
    CKS topics are developed and updated using the best available evidence. High-quality secondary evidence from NICE accredited resources - like NICE guidance and Cochrane systematic reviews - is identified first. Primary research and expert opinion is sought where necessary.
  </p>
  <h2>
    Designed for primary care professionals
  </h2>
  <p>
    CKS topics have been designed to support healthcare professionals in primary care:
  </p>
  <ul
    aria-label="Primary care audience for CKS"
  >
    <li>
      GPs
    </li>
    <li>
      GP registrars
    </li>
    <li>
      nurses
    </li>
    <li>
      pharmacists
    </li>
    <li>
      healthcare librarians
    </li>
    <li>
      medical, nursing and pharmacy students.
    </li>
  </ul>
  <p>
    Information is presented as specific clinical questions with answers and links to supporting evidence.
  </p>
  <h2>
    Always up-to-date
  </h2>
  <p>
    CKS content is continually reviewed and updated to ensure it is relevant and accurate. 65+ updates and up to 10 new topics are produced each year.
  </p>
</div>
`;
